src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+      String projectId, String namespace, String displayName) {
+    return createAppOp(
+        projectId, namespace, displayName, ANDROID_NAMESPACE_PROPERTY, ANDROID_APPS_RESOURCE_NAME);
+  }
+
+  private CallableOperation<String, FirebaseProjectManagementException> createIosAppOp(
+      String projectId, String namespace, String displayName) {
+    return createAppOp(
+        projectId, namespace, displayName, IOS_NAMESPACE_PROPERTY, IOS_APPS_RESOURCE_NAME);
+  }
+
+  private CallableOperation<String, FirebaseProjectManagementException> createAppOp(
+      final String projectId,
+      final String namespace,
+      final String displayName,
+      final String platformNamespaceProperty,
+      final String platformResourceName) {
+    return new CallableOperation<String, FirebaseProjectManagementException>() {
+      @Override
+      protected String execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/%s/%s",
+            FIREBASE_PROJECT_MANAGEMENT_URL,
+            projectId,
+            platformResourceName);
+        ImmutableMap.Builder<String, String> payloadBuilder =
+            ImmutableMap.<String, String>builder().put(platformNamespaceProperty, namespace);
+        if (!Strings.isNullOrEmpty(displayName)) {
+          payloadBuilder.put("display_name", displayName);
+        }
+        OperationResponse operationResponseInstance = new OperationResponse();
+        makePostRequest(
+            url, payloadBuilder.build(), operationResponseInstance, projectId, "Project ID");
+        if (Strings.isNullOrEmpty(operationResponseInstance.name)) {
+          throw createFirebaseProjectManagementException(
+              namespace,
+              "Bundle ID",
+              "Unable to create App: server returned null operation name.",
+              /* cause= */ null);
+        }
+        return operationResponseInstance.name;
+      }
+    };
+  }
+
+  /**
+   * An {@link ApiAsyncFunction} that transforms a Long Running Operation name to an {@link IosApp}
+   * or an {@link AndroidApp} instance by repeatedly polling the server (with exponential backoff)
+   * until the App is created successfully, or until the number of poll attempts exceeds the maximum
+   * allowed.
+   */
+  private class WaitOperationFunction implements ApiAsyncFunction<String, String> {
+    private final String projectId;
+
+    private WaitOperationFunction(String projectId) {
+      this.projectId = projectId;
+    }
+
+    /**
+     * Returns an {@link ApiFuture} that will eventually contain the App ID of the new created App,
+     * or an exception if an error occurred during polling.
+     */
+    @Override
+    public ApiFuture<String> apply(String operationName) throws FirebaseProjectManagementException {
+      SettableApiFuture<String> settableFuture = SettableApiFuture.<String>create();
+      scheduledExecutor.schedule(
```

Comment:
Expose the FirebaseApp.schedule() method via a trampoline, and use it here.

Reply:
Done.